frontend/src/scenes/insights/insightDataLogic.ts

```diff
@@ -54,7 +58,17 @@ export const insightDataLogic = kea<insightDataLogicType>([
     path((key) => ['scenes', 'insights', 'insightDataLogic', key]),
 
     connect((props: InsightLogicProps) => ({
-        values: [featureFlagLogic, ['featureFlags'], trendsLogic, ['toggledLifecycles as trendsLifecycles']],
+        values: [
+            insightLogic,
+            ['insight'],
+            featureFlagLogic,
+            ['featureFlags'],
+            trendsLogic,
+            ['toggledLifecycles as trendsLifecycles'],
+            // TODO: need to pass empty query here, as otherwise dataNodeLogic will throw
+            dataNodeLogic({ key: insightVizDataNodeKey(props), query: {} as DataNode }),
```

Comment:
You'll only then have to make sure that this specific logic is mounted before this function runs, otherwise it'll throw an error, as it should. --> props _are_ needed to actually build the logic, as a lot of reducers/selectors/etc may depend on them

Reply:
Nice! Thanks for making that work :) I'll try it right away today.